### PR TITLE
legend selection enabled

### PR DESCRIPTION
**Pull request by ahmed-Bazzara**

### CODE REVIEW

**Comment by ahmed-Bazzara** [comment]
JIRA ticket (re-opened from limitations)
https://tretton37.atlassian.net/secure/RapidBoard.jspa?rapidView=33&projectKey=QPE&modal=detail&selectedIssue=QPE-260

File: src/paint.js [hunk 1]
```diff
@@ -611,6 +611,22 @@ function setupPaint({ $, qlik }) {
                   "bordered": true,
                   "borderedHover": false
                 });
+            })
+            .on('click', function (d , i){
```
Comment (kristofferlind) [inline src/paint.js]: give d and i more descriptive names

File: src/paint.js [hunk 2]
```diff
@@ -611,6 +611,22 @@ function setupPaint({ $, qlik }) {
...
+              var tilesArray = d3.selectAll('[fill="' + colors[i] + '"]');
+              var diemensionOneElm = tilesArray[0].map(e => {
```
Comment (kristofferlind) [inline src/paint.js]: typo di_e_mensionOneElm, might aswell type out Element aswell for the name

File: src/paint.js [hunk 3]
```diff
@@ -611,6 +611,22 @@ function setupPaint({ $, qlik }) {
...
+                return e.__data__.Element1;
+              });
+              var diemensionTwoElm = tilesArray[0].map(e => {
```
Comment (kristofferlind) [inline src/paint.js]: about the same typo as above

File: src/paint.js (hunk 2 above)
Comment (kristofferlind) [inline src/paint.js]: give e a descriptive name, tile?

File: src/paint.js (hunk 3 above)
Comment (kristofferlind) [inline src/paint.js]: as above for e

File: src/paint.js [hunk 4]
```diff
@@ -598,19 +608,35 @@ function setupPaint({ $, qlik }) {
...
-              d3.selectAll('[fill="' + colors[i] + '"]')
+            .on("mouseleave", function (dataObject, index) {
+              d3.selectAll('[fill="' + colors[index] + '"]')
                 .classed({
                   "bordered": true,
                   "borderedHover": false
                 });
+            })
+            .on('click', function (d , i){
+              var rectsArray = d3.selectAll('[fill="' + colors[i] + '"]');
```
Comment (muenva) [inline src/paint.js]: Please rename `rectsArray` to something that indicates that we are dealing with the svg rectangles that we selected.
Also we need a better way to handle selection, as we are also including the `html` tag. For example if there are no hits, `rectsArray` first element will become the `html` tag. If we have some rectangles to select, then the `html` tag will always be the last. We don't want to include the `html` tag.

File: src/paint.js [hunk 5]
```diff
@@ -598,19 +608,35 @@ function setupPaint({ $, qlik }) {
...
-            .on("mouseleave", function (d, i) {
-              d3.selectAll('[fill="' + colors[i] + '"]')
+            .on("mouseleave", function (dataObject, index) {
+              d3.selectAll('[fill="' + colors[index] + '"]')
                 .classed({
                   "bordered": true,
                   "borderedHover": false
                 });
+            })
+            .on('click', function (d , i){
```
Comment (muenva) [inline src/paint.js]: +1 on giving `d` and `i` more descriptive names. Also I would add that from usability point of view we need to enable the click event on the whole legend element, not just the bar. The text should also be clickable as it has more space than the bar.

File: src/paint.js (hunk 4 above)
Comment (ahmed-Bazzara) [reply src/paint.js]: The `svgRectsArray` has the parent Node

**Comment by muenva** [comment]
As per @ahmed-Bazzara this should be closed

**Pull request closed without merging**